tests/ui/suspicious_map.stderr

```diff
@@ -1,11 +1,19 @@
 error: this call to `map()` won't have an effect on the call to `count()`
-  --> $DIR/suspicious_map.rs:4:13
+  --> $DIR/suspicious_map.rs:5:13
    |
 LL |     let _ = (0..3).map(|x| x + 2).count();
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `-D clippy::suspicious-map` implied by `-D warnings`
    = help: make sure you did not confuse `map` with `filter` or `for_each`
 
-error: aborting due to previous error
+error: this call to `map()` won't have an effect on the call to `count()`
+  --> $DIR/suspicious_map.rs:14:13
+   |
+LL |     let _ = (0..3).map(ext_closure).count();
```

Comment:
Hm, this case still gets linted...

Reply:
All we get is the name of the function. Should we do name lookup and look into the implementation to check for mutated (non-local) variables?

Which makes me thing…I suppose mutable local variables will false-negative this as well :/ . Hrm.

Reply:
Or can we do a name lookup and somehow ask if it is `FnMut`?

Reply:
I was hoping, that `maybe_body_owned_by` will get the `BodyId` of the closure and that it would Just Work™ with `mutable_variables`. Can you check if it even does get a `BodyId`?

Reply:
It does get a `BodyId`, but doesn't detect any mutated variables in it.

Reply:
So none of the examples in the test are `Upvar` instances when going through `mutated_variables`. I'm not sure what's going on here right now, but I suspect the `Copy`-ness is confusing things here?